src/dbos-runtime/config.ts

```diff
@@ -637,32 +640,33 @@ export function overwrite_config(
   }
   const poolConfig = constructPoolConfig(configFile);
 
-  if (!providedDBOSConfig.telemetry.OTLPExporter) {
-    providedDBOSConfig.telemetry.OTLPExporter = {
-      tracesEndpoint: [],
-      logsEndpoint: [],
-    };
+  const providedTraceEndpointSet = toStringSet(providedDBOSConfig.telemetry.OTLPExporter?.tracesEndpoint);
+  const configTraceEndpointSet = toStringSet(configFile.telemetry?.OTLPExporter?.tracesEndpoint);
+  for (const endpoint of configTraceEndpointSet) {
+    providedTraceEndpointSet.add(endpoint);
   }
-  if (configFile.telemetry?.OTLPExporter?.tracesEndpoint) {
-    providedDBOSConfig.telemetry.OTLPExporter.tracesEndpoint =
-      providedDBOSConfig.telemetry.OTLPExporter.tracesEndpoint?.concat(
-        configFile.telemetry.OTLPExporter.tracesEndpoint,
-      );
-  }
-  if (configFile.telemetry?.OTLPExporter?.logsEndpoint) {
-    providedDBOSConfig.telemetry.OTLPExporter.logsEndpoint =
-      providedDBOSConfig.telemetry.OTLPExporter.logsEndpoint?.concat(configFile.telemetry.OTLPExporter.logsEndpoint);
+
+  const providedLogEndpointSet = toStringSet(providedDBOSConfig.telemetry?.OTLPExporter?.logsEndpoint);
+  const configLogEndpointSet = toStringSet(configFile.telemetry?.OTLPExporter?.logsEndpoint);
+  for (const endpoint of configLogEndpointSet) {
+    providedLogEndpointSet.add(endpoint);
   }
 
-  const overwritenDBOSConfig = {
+  const overwritenDBOSConfig: DBOSConfigInternal = {
     ...providedDBOSConfig,
     name: appName,
     poolConfig: poolConfig,
-    telemetry: providedDBOSConfig.telemetry,
+    telemetry: {
```

Comment:
Also populate databaseUrl from the `DBOS_DATABASE_URL` env variable (that can be a separate PR, but we can put a note here for now).

Reply:
That's going to be the next PR where I overhaul the config file processing